Components/Sensors/BNO055.cs

```diff
@@ -0,0 +1,333 @@
+            BNO055_SIC_MATRIX_8_LSB_ADDR = 0X53,
+            BNO055_SIC_MATRIX_8_MSB_ADDR = 0X54,
+
+            /* Accelerometer Offset registers */
+            ACCEL_OFFSET_X_LSB_ADDR = 0X55,
+            ACCEL_OFFSET_X_MSB_ADDR = 0X56,
+            ACCEL_OFFSET_Y_LSB_ADDR = 0X57,
+            ACCEL_OFFSET_Y_MSB_ADDR = 0X58,
+            ACCEL_OFFSET_Z_LSB_ADDR = 0X59,
+            ACCEL_OFFSET_Z_MSB_ADDR = 0X5A,
+
+            /* Magnetometer Offset registers */
+            MAG_OFFSET_X_LSB_ADDR = 0X5B,
+            MAG_OFFSET_X_MSB_ADDR = 0X5C,
+            MAG_OFFSET_Y_LSB_ADDR = 0X5D,
+            MAG_OFFSET_Y_MSB_ADDR = 0X5E,
+            MAG_OFFSET_Z_LSB_ADDR = 0X5F,
+            MAG_OFFSET_Z_MSB_ADDR = 0X60,
+
+            /* Gyroscope Offset register s*/
+            GYRO_OFFSET_X_LSB_ADDR = 0X61,
+            GYRO_OFFSET_X_MSB_ADDR = 0X62,
+            GYRO_OFFSET_Y_LSB_ADDR = 0X63,
+            GYRO_OFFSET_Y_MSB_ADDR = 0X64,
+            GYRO_OFFSET_Z_LSB_ADDR = 0X65,
+            GYRO_OFFSET_Z_MSB_ADDR = 0X66,
+
+            /* Radius registers */
+            ACCEL_RADIUS_LSB_ADDR = 0X67,
+            ACCEL_RADIUS_MSB_ADDR = 0X68,
+            MAG_RADIUS_LSB_ADDR = 0X69,
+            MAG_RADIUS_MSB_ADDR = 0X6A
+        }
+
+        /// <summary>
+        /// The power mode of the BNO055. 
+        /// </summary>
+        private enum Powermode
+        {
+            POWER_MODE_NORMAL = 0X00,
+            POWER_MODE_LOWPOWER = 0X01,
+            POWER_MODE_SUSPEND = 0X02
+        }
+
+        /// <summary>
+        /// The operation mode of the BNO055. 
+        /// </summary>
+        public enum OperationMode
+        {
+            /* Operation mode settings*/
+            OPERATION_MODE_CONFIG = 0X00,
+            OPERATION_MODE_ACCONLY = 0X01,
+            OPERATION_MODE_MAGONLY = 0X02,
+            OPERATION_MODE_GYRONLY = 0X03,
+            OPERATION_MODE_ACCMAG = 0X04,
+            OPERATION_MODE_ACCGYRO = 0X05,
+            OPERATION_MODE_MAGGYRO = 0X06,
+            OPERATION_MODE_AMG = 0X07,
+            OPERATION_MODE_IMUPLUS = 0X08,
+            OPERATION_MODE_COMPASS = 0X09,
+            OPERATION_MODE_M4G = 0X0A,
+            OPERATION_MODE_NDOF_FMC_OFF = 0X0B,
+            OPERATION_MODE_NDOF = 0X0C
+        }
+
+        /// <summary>
+        /// The possible types of vectors. 
+        /// </summary>
+        public enum VectorType
+        {
+            VECTOR_ACCELEROMETER = Register.BNO055_ACCEL_DATA_X_LSB_ADDR,
+            VECTOR_MAGNETOMETER = Register.BNO055_MAG_DATA_X_LSB_ADDR,
+            VECTOR_GYROSCOPE = Register.BNO055_GYRO_DATA_X_LSB_ADDR,
+            VECTOR_EULER = Register.BNO055_EULER_H_LSB_ADDR,
+            VECTOR_LINEARACCEL = Register.BNO055_LINEAR_ACCEL_DATA_X_LSB_ADDR,
+            VECTOR_GRAVITY = Register.BNO055_GRAVITY_DATA_X_LSB_ADDR
+        }
+
+        int ID;
+        byte Address;
+        OperationMode Mode;
+        II2CBus I2C;
+
+        float X, Y, Z;
+

```

Comment:
My personal preference is to mark these with `private` even if it's the default just to make it as clear as possible. It also won't match the rest of the code base if these aren't specified.